module-2/backend/week_two.md

```diff
@@ -5,33 +5,71 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. At a high level, what is ActiveRecord? What does it do/allow you to do?
+  It is an ORM, allows you to use a sort of short hand to execute SQL commands.
 2. What kind of methods are `belongs_to`, and `has_many`? (i.e. class or instance) Give an example.
+  Instance methods. Using those attaches a method you can call on, on an instance of that object.
 3. What do they allow you to do?
+  They allow you to assign other objects to that object and call on them easily as a relationship.
 4. What's the difference between agile workflow and waterfall method?
+  Agile workflow builds a project with many usable stopping points, allowing you to test functionality as you go becuase 
+  you always have a functioning thing at each iteration. Waterfall saves functionality until the project is done.
 5. What is the difference between `#new` and `#create`?
+  new creates an instance of the object, create makes a new instance and saves it to the database.
 6. At a basic level, what does cURL allow you to do?
+  visit URLs from your terminal.
 7. In a database that's holding students and teachers, what will be the relationship between students and teachers? Draw the schema diagram.
+  students: id, name, (belongs to :teachers by the teacher_id); teachers: id, name, (has many :students with student_ids);
 8. Define foreign key, primary key, and schema.
+  foreign key is used to relate a database item to another database item. primary key is what a database item is primarily associated by
+  and the schema is the set of tables to describe each attribute of your database items.
 9. Describe the relationship between a foreign key on one table and a primary key on another table.
+  You might assign a foreign key to a table to associate or relate another table through its primary key.
+  like a Cars table might have a foreign key of tire_ids that would assign different primary key of a Tire object to that car.
```

Comment:
Same issue: your `tires` table would likely have a `car_id` column, not the other way around. Think of it this way:

*Preferred*
`cars_table`

|id|color|max_speed|
|-------|----|----------|
|1|blue|65|
|2|red|85|

`tires_table`

|id|position|car_id|
|----|----|----|
|1|front-left|1|
|2|front-right|1|
|3|back-left|1|
|4|back-right|1|
|5|front-left|2|
|6|front-right|2|
|7|back-left|2|
|8|back-right|2|

The alternative would be to put all the foreign keys for tires onto a car, but that would look like this:

`cars_table`

|id|color|max_speed|tire_ids|
|-------|----|-------|---|
|1|blue|65|[1, 2, 3, 4]|
|2|red|85|[5, 6, 7, 8]|

`tires_table`

|id|position|
|----|----|
|1|front-left|
|2|front-right|
|3|back-left|
|4|back-right|
|5|front-left|
|6|front-right|
|7|back-left|
|8|back-right|

Which is... not ideal (look at all those foreign key ids stuffed into a single column), and doesn't really allow us to use our relationships the way we'd like. Let me know if you'd like to discuss.